integration/mounts/mounts.go

```diff
@@ -0,0 +1,193 @@
+package mounts
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	socketPath     = "/var/run/vc/vm/foobar/console.sock"
+	kataConfigPath = "/usr/share/kata-containers/configuration.toml"
+)
+
+var (
+	kataConfigBackup = strings.Join([]string{kataConfigPath, "bak"}, ".")
+)
+
+func testOverConsole(cmd, match, socket string) bool {
+	conn, err := net.DialTimeout("unix", socket, 10*time.Second)
+	if err != nil {
+		fmt.Println("failed to connect: %w", err)
+		return false
+	}
+	defer conn.Close()
+
+	if _, err = conn.Write([]byte(cmd)); err != nil {
+		conn.Close()
+		return false
+	}
+
+	errChan := make(chan error)
+
+	go func() {
+		reader := bufio.NewReader(conn)
+		// expect echo from shell, skip what we sent and ignore first response
+		if _, err := reader.ReadString('\n'); err != nil {
+			errChan <- err
+			fmt.Println("failed to read: ", err)
+			return
+		}
+
+		if _, err = reader.ReadString('\n'); err != nil {
+			errChan <- err
+			fmt.Println("failed to read: ", err)
+			return
+		}
+
+		for {
+			response, err := reader.ReadString('\n')
+			if err != nil {
+				errChan <- err
+				fmt.Println("failed to read: ", err)
+				return
+			}
+			fmt.Println("rx:", response)
+			if strings.Contains(response, match) {
+				errChan <- nil
+			}
+		}
+	}()
+
+	select {
+	case err = <-errChan:
+		if err != nil {
+			conn.Close()
+			fmt.Println("there was an error: %w", err)
+			return false
+		}
+	case <-time.After(10 * time.Second):
+		fmt.Println("timeout trying to get response: %w", err)
+		return false
+	}
+	return true
+}
+
+func enableSandboxMounts(basedir string) error {
+	mountPath := path.Join(basedir, "test-mount")
+	filePath := path.Join(mountPath, "test-file")
+
+	mounts := strings.Join([]string{"['", mountPath, "', '", filePath, "']"}, "")
+
+	execCmd := exec.Command("crudini", "--set", kataConfigPath, "runtime", "sandbox_bind_mounts", mounts)
+	return execCmd.Run()
+}
+
+func enableDebugConsole() error {
+	execCmd := exec.Command("crudini", "--set", kataConfigPath, "hypervisor.qemu", "kernel_params", "\"agent.debug_console\"")
+	return execCmd.Run()
+}
+
+func cleanup() {
+
+	execCmd := exec.Command("ctr", "task", "kill", "foobar")
+	execCmd.Run()
+
+	execCmd = exec.Command("ctr", "task", "delete", "foobar")
+	execCmd.Run()
+
+	execCmd = exec.Command("ctr", "container", "delete", "foobar")
+	execCmd.Run()
+
+	restoreToml()
+
+}
+
+func startTestContainer() error {
+	//sudo ctr run --rm --runtime io.containerd.kata.v2 -d docker.io/library/busybox:latest foobar sh
+	execCmd := exec.Command("ctr", "run", "--rm", "--runtime", "io.containerd.kata.v2", "-d", "docker.io/library/busybox:latest", "foobar", "sh")
```

Comment:
Should we be using `quay.io` rather than `docker.io`?

/cc @devimc.

Reply:
can we use https://gallery.ecr.aws/runecast/busybox ?